Frontend/src/components/setup/admin-profile.tsx

```diff
@@ -0,0 +1,162 @@
+'use client';
+import React, { useState } from 'react';
+import {
+  Button,
+  Form,
+  Input,
+  message,
+} from 'antd';
+import { getUserDataByToken, registerAdmin, userLogin } from '@/services/auth-service';
+import useAccessToken from '@/hooks/useAccessToken';
+import { useDispatch } from 'react-redux';
+import { IUserAccessData } from '@/types/user-types';
+import { set_User_Data } from '@/redux/features/authData';
+
+const tailFormItemLayout = {
+  wrapperCol: {
+    xs: {
+      span: 24,
+      offset: 0,
+    },
+    sm: {
+      span: 16,
+      offset: 8,
+    },
+  },
+};
+
+type props = {
+  onSuccess: () => void;
+}
+
+const AdminProfile = ({onSuccess}:props) => {
+  const dispatch = useDispatch();
+  const { setSrvAccessToken, setSrvRefreshToken} = useAccessToken();
+  const [form] = Form.useForm();
+  const [loading, setLoading] = useState<boolean>(false);
+
+  const onFinish = async (values: any) => {
+    setLoading(true);
+    try{
+      const res = await registerAdmin(values.name, values.email, values.password);
+      message.loading("Creating profile", 2);
+      const tokenRes = await userLogin(values.email, values.password);
+      if(tokenRes.accessToken && tokenRes.refreshToken){
+        setSrvAccessToken(tokenRes.accessToken);
+        setSrvRefreshToken(tokenRes.refreshToken);
+      }
+      saveUserData(tokenRes.accessToken);
+      onSuccess();
+    }catch(err){
+      message.error("Failed to create profile");
+      console.log(err);
+    }
+    setLoading(false);
+  };
+
+  const saveUserData = async (token: string) => {
+    try{
+      const res = await getUserDataByToken(token).then((res) => res.data);
+      const userData: IUserAccessData = {
+        email: res.data.email,
+        fullName: res.data.fullName,
+        role: res.data.role,
+        enabled: res.data.enabled,
+        _id: res.data._id,
+      }
+      dispatch(set_User_Data(userData));
+    }catch(err){
+      console.log(err);
```

Comment:
Shouldn't we have a proper error message here?

Reply:
Like ` message.error("Failed to save user data");`

Reply:
I added a proper error message.

Reply:
Ok